client/me/help/help-courses/index.jsx

```diff
@@ -18,65 +15,31 @@ import CourseList, { CourseListPlaceholder } from './course-list';
 
 import './style.scss';
 
-class Courses extends Component {
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		this.fetchCoursesIfNeeded();
-	}
-
-	fetchCoursesIfNeeded() {
-		//TODO: When courses make it into the API we will no longer need this code.
-		//      We can move towards the use of something like <QueryHelpCourses />
-		const { courses, fetchCourses } = this.props;
-
-		if ( courses ) {
-			return;
-		}
-
-		fetchCourses();
-	}
-
-	render() {
-		const { courses, isEligible, isLoading, translate } = this.props;
-
-		return (
-			<Main className="help-courses">
-				<PageViewTracker path="/help/courses" title="Help > Courses" />
-				<HeaderCake backHref="/help" isCompact={ false } className="help-courses__header-cake">
-					{ translate( 'Courses' ) }
-				</HeaderCake>
-				{ isLoading ? (
-					<CourseListPlaceholder />
-				) : (
-					<CourseList courses={ courses } isBusinessPlanUser={ isEligible } />
-				) }
-
-				<QueryUserPurchases />
-			</Main>
-		);
-	}
-}
-
-export function mapStateToProps( state ) {
-	const purchases = getUserPurchases( state );
+function Courses() {
+	const translate = useTranslate();
+	const purchases = useSelector( getUserPurchases );
 	const purchaseSlugs = purchases && purchases.map( ( purchase ) => purchase.productSlug );
 	const isEligible =
 		purchaseSlugs &&
 		( purchaseSlugs.some( isWpComBusinessPlan ) || purchaseSlugs.some( isWpComEcommercePlan ) );
```

Comment:
Idea for a followup: this could be extracted to a local `hasEligiblePurchases` selector that returns just the boolean result. The point is that it could avoid some waster rerenders in case the `purchase` array changes, but the result doesn't.

The condition could be written in a simpler way:
```js
isEligible = purchases.some( ( { productSlug } ) =>
  isWpComBusinessPlan( productSlug ) ||
  isWpComEcommercePlan( productSlug )
);
```